src/saveEventsFile.m

```diff
@@ -146,36 +153,65 @@
 end
 
 function logFile = initializeFile(cfg, logFile)
+% This function creates the bids field structure for json files for the
+% three basic bids event columns, and for all requested extra columns. 
+%
+% Note that subfields (e.g. unit, levels etc. can be changed by the user 
+% before calling openFile.  
+
+    % initialize holy trinity (onset, trial_type, duration) columns
+    logFile(1).columns = struct( ...
+                         'onset', struct( ...
+                                         'Description', 'time elapsed since experiment start', ...
+                                         'Unit', 's'), ...
+                         'trial_type', struct( ...
+                                              'Description', 'types of trial', ...
+                                              'Levels', ''), ...
+                         'duration', struct( ...
+                                            'Description', 'duration of the event or the block', ...
+                                            'Unit', 's') ...
+                        );   
 
-    logFile = initializeStimFile(cfg, logFile);
-
-    % print the basic BIDS columns
-    fprintf(logFile(1).fileID, '%s\t%s\t%s', 'onset', 'duration', 'trial_type');
-    fprintf(1, '%s\t%s\t%s', 'onset', 'duration', 'trial_type');
 
-    printHeaderExtraColumns(logFile);
-
-    % next line so we start printing at the right place
-    fprintf(logFile(1).fileID, '\n');
-    fprintf(1, '\n');
+    logFile = initializeExtraColumns(logFile);
 
 end
 
 function logFile = initializeStimFile(cfg, logFile)
-
+    
     logFile = initializeExtraColumns(logFile);
 
+end
+
+function logFile = openFile(cfg, logFile)
+
     createDataDictionary(cfg, logFile);
 
     % Initialize txt logfiles and empty fields for the standard BIDS
-    %  event file
+    % event file
     logFile(1).fileID = fopen( ...
                               fullfile( ...
                                        cfg.dir.outputSubject, ...
                                        cfg.fileName.modality, ...
                                        logFile.filename), ...
                               'w');
 
+    if logFile(1,1).isStim == 0 
+        % print the basic BIDS columns
+        fprintf(logFile(1).fileID, '%s\t%s\t%s', 'onset', 'duration', 'trial_type');
+        fprintf(1, '%s\t%s\t%s', 'onset', 'duration', 'trial_type');
+
+        printHeaderExtraColumns(logFile);
+
+        % next line so we start printing at the right place
+        fprintf(logFile(1).fileID, '\n');
+        fprintf(1, '\n');
+        
+    elseif logFile(1,1).isStim == 1 
```

Comment:
I have changed this in a few places in the code just to make it consistent. Though now thinking back maybe it could make sense to actually have `logFile(1,1)` everywhere...

Reply:
For some reason, I thought we may get into trouble when we have additional events to save (e.g. `logFile(1,2)`, `logFile(1,3)` etc.), because the `isStim` field is not defined for them. 

But I checked again now, and even in such case `logFile(1).isStim == 0` works well and seems safe. So I think we can go with `logFile(1).isStim` as you suggest @Remi-Gau

Reply:
The reason it is set up like this is because if you "expand" a structure all the fields you do not set that already exist will be set to empty.

```matlab
>> test(1) = struct('a', 1, 'b', 1);
>> test(2).a = 2

test = 

  1×2 struct array with fields:

    a
    b

>> test(2)

ans = 

  struct with fields:

    a: 2
    b: []
```

So by forcing this to be in the first member of the structure, we make it easier for us to know where to look for that info and we prevent the need to make sure that all members of that structure have a valid `isStim` field.

Also logfile is supposed to be (n x 1) "column" structure.